Generic library documentation update

pekkaklarck opened the pull request:
Includes explaining the new `strategy:value` syntax (#908).

Still plan to update the documentation of the `__init__`, but going through individual keywords is too much work for 3.0.

pekkaklarck committed:
Re-wrote generic library document.

Includes explaining the new `strategy:value` syntax (#908).

aaltat commented on src/SeleniumLibrary/__init__.py:
```diff
@@ -42,133 +42,190 @@
+
+    Examples:
+
+    | `Click Element` | name:foo         | # Find element with class ``foo``.               |
```
I thin comment should say: 
```
Find element with name attribute ``foo``
```

pekkaklarck replied:
Yeah, I changed the example but apparently forget the comment. Will fix.

aaltat commented on src/SeleniumLibrary/__init__.py:
```diff
@@ -42,133 +42,190 @@
+    | `Click Element` | name:foo         | # Find element with class ``foo``.               |
+    | `Click Element` | default:name:foo | # Use default strategy with value ``class:foo``. |
+    | `Click Element` | //foo            | # Find element using XPath ``//foo``.            |
+    | `Click Element` | //foo            | # Use default strategy with value ``//foo``.     |
```
The locator is same in the above xpath example but the comment says something else, should the locator be just `foo` without the double forward slashes.

pekkaklarck replied:
It should be `default://foo`. Will fix.

aaltat committed:
Added missing `-character on line 205

aaltat commented on src/SeleniumLibrary/__init__.py:
```diff
@@ -42,133 +42,190 @@
-    used by all of them can be set globally using the
-    `Set Selenium Timeout` keyword. The same timeout also applies to
-    `Execute Async Javascript`.
+    used by all of them can be set globally using the `Set Selenium Timeout`
```
Should we also say that timeout can be set in the library import?

pekkaklarck replied:
Good idea. Will add.

aaltat committed:
Fixing import argument. 

'implicit_wait'  -> `implicit_wait`

pekkaklarck committed:
Tuning general docs based on review comments

pekkaklarck committed:
Type hint for IDEs

pekkaklarck committed:
Tuning general documentation.

Explain general topics in the general library documentation section,
not in __init__.
